File: src/main/java/com/E_commerce/modal/Address.java
package com.E_commerce.modal;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private String contactNumber;
    @Column(nullable = false)
    private Long pinCode;
    @Column(nullable = false)
    private String state;
    @Column(nullable = false)
    private String Area;
    private String landMark;
    private String buildingName;
    private boolean  isDefault=false;
    @ManyToOne
    @JoinColumn(name = "user_id",nullable = true)
    private User user;
}

File: src/models/Address.js
import { DataTypes, Model } from 'sequelize';
import sequelize from '../database/sequelize.js';

class Address extends Model {}

Address.init({
  id: { type: DataTypes.BIGINT.UNSIGNED, autoIncrement: true, primaryKey: true },
  userId: { type: DataTypes.BIGINT.UNSIGNED, allowNull: false },
  name: { type: DataTypes.STRING(140), allowNull: false },
  contactNumber: { type: DataTypes.STRING(30), allowNull: false },
  pinCode: { type: DataTypes.STRING(20), allowNull: false },
  state: { type: DataTypes.STRING(120), allowNull: false },
  area: { type: DataTypes.STRING(180), allowNull: false },
  landMark: { type: DataTypes.STRING(180) },
  buildingName: { type: DataTypes.STRING(180) },
  isDefault: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false }
}, {
  sequelize,
  modelName: 'Address',
  tableName: 'addresses'
});

export default Address;
